scipy/stats/_AB_SW.pyx

```diff
@@ -0,0 +1,501 @@
+            return w, pw, ifault
+
+    y = log(w1)
+    XX = log(an)
+    if n <= 11:
+        gamma = _poly(g, 2, an)
+        if y >= gamma:
+            return w, SMALL, ifault
+        y = -log(gamma - y)
+        m = _poly(c3, 4, an)
+        s = exp(_poly(c4, 4, an))
+    else:
+        m = _poly(c5, 4, XX)
+        s = exp(_poly(c6, 3, XX))
+
+    if ncens > 0:
+        ld = -log(delta)
+        bf = 1 + XX*BF1
+        Z90F = Z90 + bf * _poly(c7, 2, XX90 ** XX)**ld
+        Z95F = Z95 + bf * _poly(c8, 2, XX95 ** XX)**ld
+        Z99F = Z99 + bf * _poly(c9, 2, XX)**ld
+        ZFM = (Z90F + Z95F + Z99F)/3.
+        ZSD = (Z90*(Z90F-ZFM)+Z95*(Z95F-ZFM)+Z99*(Z99F-ZFM))/ZSS
+        ZBAR = ZFM - ZSD * ZM
+        m += ZBAR * s
+        s *= ZSD
+
+    pw = _alnorm((y-m)/s, upper)
+
+    return w, pw, ifault
+
+
+cdef double _alnorm(double x, bint upper) nogil noexcept:
+    """
+    Helper function for swilk.
+
+    Evaluates the tail area of the standardized normal curve from x to inf
+    if upper is True or from -inf to x if upper is False
+
+    Modification has been done to the Fortran version in November 2001 with the
+    following note;
+
+        MODIFY UTZERO.  ALTHOUGH NOT NECESSARY
+        WHEN USING ALNORM FOR SIMPLY COMPUTING PERCENT POINTS,
+        EXTENDING RANGE IS HELPFUL FOR USE WITH FUNCTIONS THAT
+        USE ALNORM IN INTERMEDIATE COMPUTATIONS.
+
+    The change is shown below as a commented utzero definition
+    """
+    cdef double A1, A2, A3, A4, A5, A6, A7
+    cdef double B1, B2, B3, B4, B5, B6, B7, B8, B9, B10, B11, B12
+    # cdef double utzero = 18.66
+    cdef double ltone = 7., utzero = 38., con = 1.28
+    cdef double y, z, temp
+    A1 = 0.398942280444
+    A2 = 0.399903438504
+    A3 = 5.75885480458
+    A4 = 29.8213557808
+    A5 = 2.62433121679
+    A6 = 48.6959930692
+    A7 = 5.92885724438
+    B1 = 0.398942280385
+    B2 = 3.8052e-8
+    B3 = 1.00000615302
+    B4 = 3.98064794e-4
+    B5 = 1.98615381364
+    B6 = 0.151679116635
+    B7 = 5.29330324926
+    B8 = 4.8385912808
+    B9 = 15.1508972451
+    B10 = 0.742380924027
+    B11 = 30.789933034
+    B12 = 3.99019417011
+    z = x
+    if not (z > 0):  # negative of the condition to catch NaNs
+        upper = False
+        z = -z
+    if not ((z <= ltone) or (upper and z <= utzero)):
+        return 0. if upper else 1.
+    y = 0.5 * z * z
+    if z <= con:
+        temp = 0.5 - z*(A1 - A2*y/(y + A3 - A4/(y + A5 + A6/(y + A7))))
+    else:
+        temp = B1 * exp(-y)/(z - B2 + B3/(z + B4 + B5/(z - B6 + B7/
+                             (z + B8 - B9/(z + B10 + B11/(z + B12))))))
+
+    return temp if upper else (1-temp)
+
+
+cdef double _ppnd(double p) noexcept:
+    """
+    Helper function for swilk. Unused ifault return is
+    commented out from return statements.
+    """
+    cdef double A0, A1, A2, A3, B1, B2, B3, B4, C0, C1, C2, C3, D1, D2
+    cdef double q, r, split = 0.42, temp
+    # cdef int ifault = 0
```

Comment:
That's for someone who wants to enable ifault later at some point, I basically disabled it since we didn't use. But left it as a comment at the top uncomment if needed.